src/utils/image.ts

```diff
@@ -0,0 +1,89 @@
+interface Dimensions {
+  width: number
+  height: number
+}
+
+/**
+ * Get the dimensions of the image after resize
+ *
+ * @param imgWidth  Current image width
+ * @param imgHeight Current image height
+ * @param maxWidth  Desired max width
+ * @param maxHeight Desired max height
+ *
+ * @returns Downscaled dimensions of the image to fit in the bounding box
+ */
+export function getDimensions(imgWidth: number, imgHeight: number, maxWidth?: number, maxHeight?: number): Dimensions {
+  const ratioWidth = maxWidth ? imgWidth / maxWidth : 1
+  const ratioHeight = maxHeight ? imgHeight / maxHeight : 1
+
+  const ratio = Math.max(ratioWidth, ratioHeight)
+
+  // No need to resize
+  if (ratio <= 1) return { width: imgWidth, height: imgHeight }
+
+  return { width: imgWidth / ratio, height: imgHeight / ratio }
+}
+
+/**
+ * Resize image passed to fit in the bounding box defined with maxWidth and maxHeight.
+ * Note that one or both of the bounding box dimensions may be omitted
+ *
+ * @param file      Image file to be resized
+ * @param maxWidth  Maximal image width
+ * @param maxHeight Maximal image height
+ *
+ * @returns Promise that resolves into the resized image blob
+ */
+export function resize(file: File, maxWidth?: number, maxHeight?: number): Promise<Blob> {
+  return new Promise((resolve, reject) => {
+    const allowedTypes = [
+      'image/bmp',
+      'image/gif',
+      'image/vnd.microsoft.icon',
+      'image/jpeg',
+      'image/png',
+      'image/svg+xml',
+      'image/tiff',
+      'image/webp',
+    ]
+
+    if (!file.size || !file.type || !allowedTypes.includes(file.type)) reject('File not supported!')
```

Comment:
Is this missing a `return`? I think `reject` on its own does not stop execution

Reply:
You are very right, this needs to have `return`